src/github/repository.ts

```diff
@@ -3,15 +3,30 @@ import { emitWebhookProcessedMetrics } from "utils/webhook-utils";
 export const deleteRepository = async (context, jiraClient): Promise<void> => {
 	context.log(`Deleting dev info for repo ${context.payload.repository?.id}`);
 
-	const jiraResponse = await jiraClient.devinfo.repository.delete(
+	const deleteDevinfoPromise = jiraClient.devinfo.repository.delete(
 		context.payload.repository?.id
 	);
+
+	const deleteDeploymentsPromise = jiraClient.deployment.delete(
+		context.payload.repository?.id
+	);
+
+	const deleteBuildsPromise = jiraClient.workflow.delete(
+		context.payload.repository?.id
+	);
+
+	await Promise.all([
```

Comment:
Do we have a test for this?

Reply:
How about transaction/retry mechanism? Are bellow the intended mechanism?
1. This msg is taken from sqs, so assuming this call error, the msg will `stay` in sqs for retry (up to ? times)
2. The deletion is idempotent, so if some call failed, some calls success, it is okay, next retry is safe to call whole thing again (like the success calls, say `deleDevinfoPromise` won't error if called again with a deleted `id`)